### PR TITLE
rhn_register: fix broken imports, add unit tests

### PR DESCRIPTION
**Pull request by pilou-**
##### SUMMARY
Fix [broken import](https://github.com/ansible/community/wiki/Testing:-progress-tracker#fixing-broken-imports) in `rhn_register` module due to Python 3 incompatibility and minor changes and update requirements documentation.

Following updates initially included in this pull request are now included in #27533 in order to facilitate reviews:
- ~~improve error message when requirement is missing~~
- ~~call `logout`~~

Removed:
- ~~use `required_*`~~

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
packaging/os/rhn_register.py

##### ANSIBLE VERSION

```
ansible-playbook 2.4.0 (devel 30ad30c470) last updated 2017/07/17 03:07:50 (GMT +200)
```

### CODE REVIEW

**Comment by pilou-** [comment]
@jtyr @sysadmind (as randomly selected maintainer of `packaging/os` namespace) could you please review this pull request ? At least two `shipit` comments (from maintainers or namespace maintainers or core team members) are required in order to [automerge](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md#community) this pull request.

File: lib/ansible/modules/packaging/os/rhn_register.py
Comment (mkrizek) [inline lib/ansible/modules/packaging/os/rhn_register.py]: Am I missing something or is this supposed to be `self.server = xmlrpc_client.ServerProxy(url)`?

File: lib/ansible/modules/packaging/os/rhn_register.py
Comment (sysadmind) [inline lib/ansible/modules/packaging/os/rhn_register.py]: I believe this is a yaml array. The hyphen up front indicates an array element. I believe the new requirements should be on a new line.

https://github.com/ansible/ansible/blob/devel/examples/DOCUMENTATION.yml#L31

**Review by sysadmind — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/packaging/os/rhn_register.py
Comment (pilou-) [reply lib/ansible/modules/packaging/os/rhn_register.py]: Oups, fixed in https://github.com/ansible/ansible/pull/26878/commits/b88ca8845e0b3c8c5dd92c9dcb2ba6c40d63aca5.

File: lib/ansible/modules/packaging/os/rhn_register.py
Comment (pilou-) [reply lib/ansible/modules/packaging/os/rhn_register.py]: Indeed, fixed in https://github.com/ansible/ansible/pull/26878/commits/4eab55e532560dd7c4112350d9c857af704432cd.

**Comment by pilou-** [comment]
Pull request rebased, both comments have been addressed in separate commits. I have added unit tests in another commit.

ready_for_review

**Comment by mattclay** [comment]
Closing and re-opening to trigger CI.

**Comment by pilou-** [comment]
pull-request updated:
- PEP 8 fixed in unit test
- drop support for Python 2.4

**Comment by pilou-** [comment]
Not sure why this error is detected here, I created a pull request which remove non breaking space in docs/docsite/rst/roadmap/ROADMAP_2_2.rst.

**Comment by pilou-** [comment]
Last commit updated in order to trigger the CI.

**Comment by pilou-** [comment]
CI found some Python 3 incompatibility.

**Comment by alikins** [comment]
@pilou- https://github.com/ansible/ansible/compare/devel...alikins:pr-26878-py3-mock has some fixes for py3 and Rhn.enable()

**Review by alikins — APPROVED** [review]
LGTM, pending fixing failing py3 tests

**Comment by pilou-** [comment]
Pull request rebased and updated:
- @alikins you're right, `Rhn.enable` should be mocked
- update units test:
  - XMLRPC `logout` is called now
  - ensure orders of the XMLRPC calls

**Review by mkrizek — APPROVED** [review]

**Comment by pilou-** [comment]
Get rid of the `needs_revision` label: ready_for_review.

@alikins @mkrizek `approved` pull request status is ignored by the bot, [`shipit`](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md#cmd-shipit) command must be used in order to approve a pull request: could you add a comment containing `shipit`? Then:
1. the `shipit` label should automatically be added
2. a core team member would be able to review and potentially merge this pull-request.

**Comment by alikins** [comment]
shipit :shipit:

**Pull request merged**
